### PR TITLE
Save the date reference on register using the provided global/window context instead of on require using the default available context

### PR DESCRIPTION
**Pull request by roland-vachter**
This library has great potential for testing with different timezones, however it would help if the library would be interoperable with other Date manipulation libraries. Therefore on register, it would be much helpful if the original reference to the Date object would be saved on runtime (call to register) and from the provided global context, instead of on require from the default available context.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -21,9 +21,15 @@ var date_with_offset = /^\d\d\d\d-\d\d-\d\d \d\d:\d\d:\d\d(\.\d\d\d)? (Z|(-|\+|)
 var date_rfc_2822_regex = /^\d\d-\w\w\w-\d\d\d\d \d\d:\d\d:\d\d (\+|-)\d\d\d\d$/;
 var local_date_regex = /^\d\d\d\d-\d\d-\d\d[T ]\d\d:\d\d(:\d\d(\.\d\d\d)?)?$/;
 
+var currentTime = null;
+
 function MockDate(param) {
   if (arguments.length === 0) {
-    this.d = new _Date();
+    if (typeof currentTime === 'number') {
+        this.d = new _Date(currentTime);
```
Comment (Jimbly) [inline index.js:29]: In reality, the current time is not constant, and code often has subtle bugs resulting from assuming two nearby `Date.now()` calls return the same value... would `new _Date(currentTime++)` be a better choice (with README saying `starting from a given timestamp`)?

Comment (roland-vachter) [reply index.js:29]: @Jimbly What I actually need is what cypress does with clock (but using that feature there is no timezone simulation): https://docs.cypress.io/api/commands/clock#Examples
So what do you think about having a `tick` function with a value for increasing the time?
The problem with `currentTime++` is that it's not controlled, and in a test it's important to know what to expect. It's hard to follow `currentTime++` so an assert would be impossible to be made.

Comment (Jimbly) [reply index.js:29]: An assert could be made, but it, like any safe code dealing with elapsed time, might need to be checking a range instead of an exact value ^_^.  Understandably possibly harder to follow though.  A tick function is not unreasonable, however, unlike that Cypress example, it's not going to affect things like setInterval, so maybe a little confusing.  If you want to use Cypress, and mock timezones, you should just be able to use them both - as long Cypress mocks the `Date` object _before_ you require/import `timezone-mock`, then `timezone-mock` will use Cypress's mocked Date, and they should work together.  A simple change to timezone-mocks's `register()` to move the `_Date` initialization to there (`if (Date !== MockDate) { _Date = Date; }`) would make that even easier (just the order of registering/mocking would matter, not importing).

Comment (roland-vachter) [reply index.js:29]: That unfortunately doesn't work out of the box, as `cy.clock()` changes the Date object too late, when this library already saved a reference to the `Date` object.
In order to work, and also to give a bit of a freedom with combining other Date manipulation libraries would be to save the reference to the `Date` object in this library on `register` (`_Date = Date`).
I think this change would give much more flexibility. What do you think?

**Comment by roland-vachter** [comment]
Updated the PR. @Jimbly can you please have a look?

File: index.js [hunk 2]
```diff
@@ -288,3 +292,14 @@ function unregister(glob) {
   glob.Date = _Date;
 }
 exports.unregister = unregister;
+
+
+function setCurrentTime(timeToSet) {
+	currentTime = timeToSet;
+}
+exports.setCurrentTime = setCurrentTime;
```
Comment (Jimbly) [inline index.js:300]: These functions aren't needed anymore?

File: index.js [hunk 3]
```diff
@@ -203,9 +203,9 @@ MockDate.prototype.toString = function () {
   return str.join('');
 };
 
-MockDate.now = _Date.now;
+MockDate.now = () => _Date.now;
 
-MockDate.UTC = _Date.UTC;
+MockDate.UTC = () => _Date.UTC;
```
[the remark this answers is not in the document]
Comment (roland-vachter) [reply index.js]: Sorry for that. Indeed, arrow function tricked me :)
Pushed a fix for that.

File: index.js (hunk 2 above)
Comment (roland-vachter) [reply index.js]: Yes you're right, removed them. Thanks for double checking!

**Review by Jimbly — APPROVED** [review]

**Comment by Jimbly** [comment]
Published as `timezone-mock@1.3.3`
